### DISCUSSION TITLE
Changing the Workspace Environment at Runtime.

### DISCUSSION THREAD

**Discussion by EndlessTrax** [microsoft/vscode-discussions#1005] [Extension Development QnA] [answered]
I'm looking to write an extension that will replace environment variables that contain a predetermined reference value with an actual value stored in a vault/keyring. The goal is to never have the real value saved to the file or accidentally committed. A common problem you've probably all encountered.  I know I can set values in the workspace file, but this runs into the same problems.

Does VS Code allow extensions to alter the environment and inject environment variables into it **before** the process is started to run an application? If so, can you point me to the relevant API documentation that seems to be alluding me?

**Comment by r3m0t** [comment, marked as answer]
https://code.visualstudio.com/api/references/vscode-api#ExtensionContext - `environmentVariableCollection`
